### ISSUE TITLE
Wildcard hostnames in Apaches VirtualHosts aren't recognised

### ISSUE THREAD

**Issue by osirisinferi** [certbot/certbot#8511]
The WONTFIX of #3821 from 2016 has become an issue again, already some years now by the way. Guess it's not that common.

It seems the Apache certbot plugin doesn't recognise wildcard hostnames.

For example, for:

```
(...)
ServerName example.com
ServerAlias *.example.com
(...)
```

certbot would only suggest `example.com` when ran without any `-d` options.

As wildcards are nowadays supported by Let's Encrypt, I'd like to re-issue this.

**Comment by ohemorange** [comment]
Well, but the Apache plugin doesn't handle wildcard certs, those are handled by the DNS plugins. So we still can't actually do anything about detecting wildcard certs when suggesting domains found by the authenticator...

In the case where the authenticator is set to a dns plugin and the installer is set to a apache, then yes, we could, but currently domain detection is done by the authenticator, not the installer. So it'd be a bit more complicated of a fix.

**Comment by osirisinferi** [comment]
> In the case where the authenticator is set to a dns plugin and the installer is set to a apache, then yes, we could, but currently domain detection is done by the authenticator, not the installer. So it'd be a bit more complicated of a fix.

When I run `certbot -a manual -i apache`, I get the list of domains from Apache on my screen? Or am I misinterpreting your domain detection explanation here?

**Comment by ohemorange** [comment]
Ah, nice catch. I feel like there was a recent issue discussing this exact thing that proposed changing it, which may be what I was thinking of. I'd have to check where the domains are filtered out; that might need to be moved from the plugin to core certbot, but it'd certainly be more doable. I wonder how often people use manual auth and apache installer though? That's not rhetorical.

**Comment by alevinetx** [comment]
I don't use manual auth, but do use r53 with apache.

Is this the right ticket for the "Which VirtualHosts would you like to install the wildcard certificate for?" issue when using vhosts and non-interactive?

```
        ServerName myserver.com
        ServerAlias *.myserver.com
```

run -n
```
2021-05-17 15:20:49,216:ERROR:certbot._internal.log:Missing command line flag or config entry for this setting:
Which VirtualHosts would you like to install the wildcard certificate for?
File: /etc/httpd/conf.d/myserver.conf
Addresses: *:443
Names: myserver.com, *.myserver.com
HTTPS: Yes
```

I thought ```--apache-vhost-root myserver.com``` would do the trick, but it seems that switch does not do what I hoped.

**Comment by osirisinferi** [comment]
@alevinetx Running certbot **non**-interactive would require the use of the `-d` option to specify the required hostnames in your certificate.  The `--apache-vhost-root` option is not meant for hostnames, but is an option for the directory where certbot will place the generated virtualhost configuration files. I think. The documentation of that option is, well, poor at best. I'm not sure you thought to achieve with your command though? Is the `-d` option not sufficient?

@ohemorange 
> I wonder how often people use manual auth and apache installer though? That's not rhetorical.

That was just for testing purposes, but as seen above other combinations with DNS plugins are possible *and* very logical.

**Comment by alevinetx** [comment]
@osirisinferi  No, the ```-d``` option gives the messages above.  My run command was assuredly abbreviated, but this is full form:

```certbot run --dns-route53  -d myserver.com -d *.myserver.com  -i apache -n --agree-tos```

There are other VirtualHosts defined in Apache, and they have their own certbot calls.

**Comment by osirisinferi** [comment]
@alevinetx I can't reproduce your error with the exact same command you've used. Please open either a new issue here on Github or perhaps even better open a thread in the [#help](https://community.letsencrypt.org/c/help/13) section of our [Community](https://community.letsencrypt.org/).

**Comment by alevinetx** [comment]
@osirisinferi  Very odd.

Are you using the snap version, etc?

```
[root@dmz ~]# snap list
Name                 Version   Rev    Tracking       Publisher     Notes
certbot              1.15.0    1150   latest/stable  certbot-eff✓  classic
certbot-dns-route53  1.15.0    730    latest/stable  certbot-eff✓  -
core                 16-2.50   11081  latest/stable  canonical✓    core
core20               20210429  1026   latest/stable  canonical✓    base
[root@dmz ~]# snap version
snap    2.49-2.el8
snapd   2.49-2.el8
series  16
centos  8
kernel  4.18.0-240.15.1.el8_3.x86_64
```

**Comment by osirisinferi** [comment]
I believe it's still relevant.

**Comment by osirisinferi** [comment]
Still relevant, the `apache` plugin still doesn't recognise wildcard `ServerName`/`ServerAlias` values.

**Comment by zoracon** [comment]
For now we decided to leave this unplanned and close. But if there's more interest or concerns from more users we can certainly reconsider reopening this issue.
